modules/custom/openy_system/src/OpenyModulesManager.php

```diff
@@ -0,0 +1,246 @@
+              $parent_entity->set($parent_field_name, $clean_field_data);
+              $parent_entity->save();
+            }
+          }
+        }
+      }
+      else {
+        try {
+          $storage_handler->delete($entities);
+        }
+        catch (\Exception $e) {
+          watchdog_exception('openy', $e, "Error! Can't delete content related 
+          to @entity_type bundle <b>@bundle</b>, please @removeContent (Entity ID's - @ids).<br>
+          After this @removeBundle @config_entity_type bundle <b>@bundle</b>.", [
+            '@entity_type' => $content_entity_type,
+            '@config_entity_type' => $config_entity_type,
+            '@bundle' => $bundle,
+            '@ids' => implode(', ', $ids),
+            '@removeContent' => $this->getLink($content_entity_type, 'remove it manually'),
+            '@removeBundle' => $this->getLink($config_entity_type, 'delete config entity', $bundle),
+          ], RfcLogLevel::NOTICE);
+          $operation_success = FALSE;
+        }
+      }
+    }
+    return $operation_success;
+  }
+
+  /**
+   * Helper function for ling generation.
+   *
+   * @param string $entity_type
+   *   Config or content entity type.
+   * @param string $link_text
+   *   Text that will be displayed inside link.
+   * @param string $bundle
+   *   For config entity type need to set bundle (it used in delete link).
+   *
+   * @return bool|string
+   *   Rendered html link or FALSE.
+   */
+  public function getLink($entity_type, $link_text = 'here', $bundle = NULL) {
+    $url = FALSE;
+    switch ($entity_type) {
+      case 'node':
+        $url = Url::fromUserInput('/admin/content');
+        break;
+
+      case 'node_type':
+        $url = Url::fromRoute('entity.node_type.delete_form', ['node_type' => $bundle]);
+        break;
+
+      case 'block_content':
+        $url = Url::fromRoute('entity.block_content.collection');
+        break;
+
+      case 'block_content_type':
+        $url = Url::fromRoute('entity.block_content_type.delete_form', ['block_content_type' => $bundle]);
+        break;
+
+      case 'media':
+        // TODO: Replace url and route after switching to core media.
+        $url = Url::fromUserInput('/admin/content/media');
+        break;
+
+      case 'media_bundle':
+        // TODO: Replace url and route after switching to core media.
+        $url = Url::fromUserInput("/admin/structure/media/manage/$bundle/delete");
+        break;
+
+      case 'webform_submission':
+        $url = Url::fromRoute('entity.webform_submission.collection');
+        break;
+
+      case 'webform':
+        $url = Url::fromUserInput('/admin/structure/webform');
+        break;
+
+      case 'paragraphs_type':
+        $url = Url::fromRoute('entity.paragraphs_type.delete_form', ['paragraphs_type' => $bundle]);
+        break;
+    }
+
+    if (!$url) {
+      return FALSE;
+    }
+
+    $link = Link::fromTextAndUrl($link_text, $url)->toRenderable();
+    return render($link);
```

Comment:
Wouldn't be better to make this as a renderable array to let the system render it on final stages?
Drupal 8 has '#type' => 'link'

Reply:
Let's work on this as a follow up